include/swift/AST/NameLookup.h

```diff
@@ -103,15 +103,18 @@ struct LookupResultEntry {
   /// extension (if it found something at that level).
   DeclContext *BaseDC;
 
+  /// The declaration that defines the base of the call to `Value`
+  ValueDecl *BaseDecl;
```

Comment:
Could you please add a comment that explains at what conditions the `BaseDecl` wouldn't be available?

Reply:
Should it always be computed and stored instead of logic being split between `LookupResultEntry::getBaseDecl()` and lookup itself? It seems like it might be a simpler setup...

Reply:
Good idea, I moved the remaining logic from `LookupResultEntry::getBaseDecl()` into the implementation of `ASTScope::lookupUnqualified`. Now this `BaseDecl` property is always non-null, as long as `BaseDC` is available.